test/test_torch.py

```diff
@@ -6383,6 +6384,119 @@ def test_clamp(self, device):
         with self.assertRaisesRegex(RuntimeError, error_msg):
             m1.clamp_()
 
+    def test_clamp_with_tensors(self, device):
+        m1 = torch.rand(100, device=device).mul(5).add(-2.5)  # uniform in [-2.5, 2.5]
+        # just in case we're extremely lucky.
+        min_vals = torch.linspace(-1, 0, 100, device=device)  # [-1..0]
+        max_vals = torch.linspace(1, 0, 100, device=device)  # [1..0]
+
+        res1 = m1.clone()
+        res1.clamp_(min_vals, max_vals)
+        res2 = m1.clone()
+        for i in iter_indices(res2):
+            res2[i] = max(min_vals[i], min(max_vals[i], res2[i]))
+        self.assertEqual(res1, res2)
+
+        out = m1.clone()
+        torch.clamp(m1, min=min_vals, max=max_vals, out=out)
+        self.assertEqual(out, res1)
+
+        res1 = torch.clamp(m1, min=min_vals)
+        res2 = m1.clone()
+        for i in iter_indices(res2):
+            res2[i] = max(min_vals[i], res2[i])
+        self.assertEqual(res1, res2)
+
+        torch.clamp(m1, min=min_vals, out=out)
+        self.assertEqual(out, res1)
+
+        res1 = torch.clamp(m1, max=max_vals)
+        res2 = m1.clone()
+        for i in iter_indices(res2):
+            res2[i] = min(max_vals[i], res2[i])
+        self.assertEqual(res1, res2)
+
+        torch.clamp(m1, max=max_vals, out=out)
+        if (out - res1).abs().max() > 0.01:
```

Comment:
Is this debug code? That shouldn't be included in the PR.

Reply:
Yes, that was debug code, since I used the CI for debugging CUDA. It is now removed.